### PR TITLE
[DEVHAS-133] Sanitize Error Messages

### PR DESCRIPTION
**Pull request by kim-tsao**
Signed-off-by: Kim Tsao <ktsao@redhat.com>

### What does this PR do?:

- Creates a set of custom gitops error messages so they are sanitized before returning to the client
- Update similarly worded error messages so they are consistent and can be reused by the same custom error message
- Update unit tests
- Added a new Fuzz test for the util function 

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-133

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]

File: pkg/errors.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+type GitGenResourcesAndOverlaysError struct {
+	path          string
+	componentName string
+	err           error
+	cmdType       GitCmd
+}
+
+func (e *GitGenResourcesAndOverlaysError) Error() string {
+	failedToGenerate := "failed to generate the gitops resources in "
+	if e.cmdType == genOverlays {
```
Comment (keithchong) [inline pkg/errors.go:87]: Could genOverlays be "generate overlays directory"
and then  we don't need this if block, and modify the message on line 90 to be:
"failed to %q for component %q: %s" 

That way the message on line 90 is similar to the other messages like on line 75, and 102, etc.?  They all start with "failed to...."

File: pkg/errors.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+	// Check to see what gitops actions have take place in order to append the correct preposition
+	cmdMsg := e.cmdType
+	if e.cmdType == checkGitDiff {
+		cmdMsg = cmdMsg + " in"
+	} else if e.cmdType == commitFiles || e.cmdType == pushRemote {
+		cmdMsg = cmdMsg + " to"
+	} else if e.cmdType == getCommitID {
+		cmdMsg = cmdMsg + " for"
+	}
```
Comment (keithchong) [inline pkg/errors.go:59]: How about addComponents?

**Review by keithchong — CHANGES_REQUESTED** [review]
Hi Kim, added a couple of comments.

File: pkg/errors.go (hunk 1 above)
Comment (kim-tsao) [reply pkg/errors.go:87]: yeah, I did start off with a similar solution but the substitution leaves an extra space if it's not an "overlays dir".  This breaks unit testing

File: pkg/errors.go (hunk 2 above)
Comment (kim-tsao) [reply pkg/errors.go:59]: good catch, I'll fix this

**Review by keithchong — APPROVED** [review]
LGTM

**Pull request merged**